gh-101578: Fixup NEWS and add What's New entry for new exception APIs (#102157)
python/cpython: Doc/whatsnew/3.12.rst

  get a frame variable by its name.
  (Contributed by Victor Stinner in :gh:`91248`.)

> * Add :c:func:`PyErr_GetRaisedException` and :c:func:`PyErr_SetRaisedException`
>   for saving and restoring the current exception.
>   These functions take and return a single exception,

Reviewer: ```suggestion
* Add :c:func:`PyErr_GetRaisedException` and :c:func:`PyErr_SetRaisedException`
  for saving and restoring the current exception.
  These functions return and accept a single exception object,
```

* Clarify that we're talking about a single exception object (or whatever is an appropriate word to describe an instance of the PyException type as opposed to an exception as a higher-level construct.), as the current text implies that it previously accepted/returned multiple _exceptions_ rather a tuple of multiple objects that comprised one exception
* The current text implies both functions both accept and return an exception
* The functions are listed out of order to the verbs that respectively apply to them.
* Could use more idiomatic/conventional language (accept vs. take)

Author: That's a nice improvement; thanks!

>   rather than the triple arguments of the now deprecated

Reviewer: ```suggestion
  rather than the triple arguments of the now-deprecated
```

Author: TIL!

  get a frame variable by its name.
  (Contributed by Victor Stinner in :gh:`91248`.)

* Add :c:func:`PyErr_GetRaisedException` and :c:func:`PyErr_SetRaisedException`
  for saving and restoring the current exception.
  These functions return and accept a single exception object,
  rather than the triple arguments of the now-deprecated
  :c:func:`PyErr_Fetch` and :c:func:`PyErr_Restore`.

>   This is less error prone and a bit more efficient.

Reviewer 2: (Contributed by Mark Shannon in GH-101578)

>   Also add :c:func:`PyException_GetArgs` and :c:func:`PyException_SetArgs`

Reviewer 2: ```suggestion
  Add :c:func:`PyException_GetArgs` and :c:func:`PyException_SetArgs`
```

  as convenience functions for retrieving and modifying

>   the :attr:`~BaseException.args` passed to the exception's constructor.

Reviewer 2: (Contributed by Mark Shannon in https://github.com/python/cpython/issues/101578)

  (Contributed in :gh:`47146` by Petr Viktorin, based on
  earlier work by Alexander Belopolsky and Matthias Braun.)

* :c:func:`PyErr_Fetch` and :c:func:`PyErr_Restore` are deprecated.
  Use :c:func:`PyErr_GetRaisedException` and

>   :c:func:`PyErr_SetRaisedException` instead.

Reviewer 2: (Contributed by Mark Shannon in https://github.com/python/cpython/issues/101578)
